sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -325,14 +325,37 @@ public String getAccountUrl() {
         ListBlobContainersOptions options, Duration timeout) {
         options = options == null ? new ListBlobContainersOptions() : options;
 
-        ListBlobContainersIncludeType includeType = options.getDetails().toIncludeType();
         return StorageImplUtils.applyOptionalTimeout(
             this.azureBlobStorage.services().listBlobContainersSegmentWithRestResponseAsync(
                 options.getPrefix(), marker, options.getMaxResultsPerPage(),
-                includeType == null ? null : Collections.singletonList(includeType),
+                toIncludeTypes(options.getDetails()),
                 null, null, Context.NONE), timeout);
     }
 
+    /**
+     * Converts {@link BlobContainerListDetails} into list of {@link ListBlobContainersIncludeType}
+     * that contains only options selected. If no option is selected then null is returned.
+     *
+     * @return a list of selected options converted into {@link ListBlobContainersIncludeType}, null if none
+     * of options has been selected.
+     */
+    private List<ListBlobContainersIncludeType> toIncludeTypes(BlobContainerListDetails blobContainerListDetails) {
```

Comment:
Why is this here now instead of on the Details type? Is it so we don't have to have a public method on Details to allow access across packages?

Reply:
This is the result of discussion on the APIView. We don't expect customer to call this and have option to hide it as there is only one place this is being used.